From 03d5654a16cce44bf449ef2871a16f5c6bcb6c14 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Thu, 9 May 2013 18:43:47 +0100
Subject: [PATCH 1/5] Fix #2324 : Encryption level set by deluge does not match
 libtorrent values

The clients are using range (0-2) whereas actual bit values are (1-3)
---
 deluge/core/preferencesmanager.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/deluge/core/preferencesmanager.py b/deluge/core/preferencesmanager.py
index 7a3cc2c..02665a7 100644
--- a/deluge/core/preferencesmanager.py
+++ b/deluge/core/preferencesmanager.py
@@ -345,11 +345,12 @@ def _on_set_utpex(self, key, value):
 
     def _on_set_encryption(self, key, value):
         log.debug("encryption value %s set to %s..", key, value)
+        pe_enc_level = {0: lt.enc_level.plaintext, 1: lt.enc_level.rc4, 2: lt.enc_level.both}
         pe_settings = lt.pe_settings()
         pe_settings.out_enc_policy = \
             lt.enc_policy(self.config["enc_out_policy"])
         pe_settings.in_enc_policy = lt.enc_policy(self.config["enc_in_policy"])
-        pe_settings.allowed_enc_level = lt.enc_level(self.config["enc_level"])
+        pe_settings.allowed_enc_level = lt.enc_level(pe_enc_level[self.config["enc_level"]])
         pe_settings.prefer_rc4 = self.config["enc_prefer_rc4"]
         self.session.set_pe_settings(pe_settings)
         set = self.session.get_pe_settings()
-- 
1.8.3.1

